enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,652 @@
+
+The insights-runtime-extractor functionality is provided by an image that must be pullable from the `openshift-insights` namespace and present in the cluster at installation.
+
+#### Variation and form factor considerations [optional]
+
+This enhancement to the Insight Operator is not impacting the Control Plane of OpenShift and should be available regardless of the form of OpenShift (Standalone OCP,  Microshift, Hypershift).
+
+This enhancement only gathers runtime information from Linux worker nodes and will not touch Windows worker nodes. The DaemonSet will be configured to only be deployed on Linux worker nodes (having the label `beta.kubernetes.io/os=linux`).
+
+### API Extensions
+
+This proposal does not require any API Extensions to the OpenShift Insights Operator.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+#### Hypershift [optional]
+
+This proposal only aims to gather data about the Data plane and will not inspect any containers running in the Control Plane.
+
+#### Standalone Clusters
+
+This proposal is relevant for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal affects single-node deployment as there is additional resource consumption
+when the Insights Operator gathers workload data that will affect the only worker node.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Container Scanner API
+
+The Insights Operator communicates with the insights-runtime-extractor pods deployed by its DaemonSet with a simple HTTP server that has the following API:
+
+* `GET /gather-runtime-info`
+  * Inspect the all the containers running on the worker node. The HTTP response returns a JSON payload with the extracted data organized in a tree of `namespace/pod-name/container-id/runtime-info`:
+
+```json
+{
+  "my-namespace": {
+    "my-pod-1": {
+      "cri-o://xxxx": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "8.8",
+        "runtime-kind": "Java",
+        "runtime-kind-version": "17.0.8",
+        "runtime-kind-implementer": "Red Hat, Inc.",
+        "runtimes": [
+          {
+            "name": "Quarkus",
+            "version": "3.4.1",
+          }
+        ],
+      },
+      "cri-o://yyyy": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "9.2",
+      }
+    },
+    "my-pod-2": {
+      [...]
+    }
+  },
+  "other-namespace": {
+    "other-pod-1": {
+      "cri-o://zzzz": {
+        [...]
+      }
+    }
+  }
+}
+```
+
+#### insights-runtime-extractor Implementation
+
+Inspecting a container is achieved by running on the same worker node that the inspected container and looking at the host process table. The insights-runtime-extractor enters the container’s process namespaces to extract meaningful information about the runtime stack.
```

Comment:
Can we clarify here that it will just join the namespaces and *not* the cgroups so it doesn't count towards container limits?

Reply:
done in https://github.com/openshift/enhancements/blob/e083208a087afcc9745bd0c0e154174985e16c76/enhancements/insights/insights-runtime-extractor.md#insights-runtime-extractor-constraints